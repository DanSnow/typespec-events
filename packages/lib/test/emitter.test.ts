import { expect, it } from 'vitest';
import { emit } from './test-host.js';

it('emit events.zod.ts', async () => {
  const results = await emit('@event("cta_clicked") model CtaClicked { title: string}');
  expect(results['events.zod.ts']).toMatchSnapshot();
});
