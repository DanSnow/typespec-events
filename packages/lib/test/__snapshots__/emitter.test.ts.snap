// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`emit events.zod.ts 1`] = `
"import { z } from 'zod';

export const cta_clickedSchema = z.object({ title: z.string() });"
`;
